### PR TITLE
Issue #6018: Add to CI import order validation

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6018 

CI for import order checks  for `guava(v25)` and `java design patterns`

### CODE REVIEW

**Comment by rnveach** [comment]
@romani Issue isn't approved.

**Comment by romani** [comment]
I didn't approved the issue with idea to let @rnveach (second opinion) review it and approve to make sure I am not going crazy. 

I will approve issue, but there are problems in PR, I will give feedback a bit later.

File: .ci/wercker.sh [hunk 1]
```diff
@@ -276,6 +276,32 @@ no-exception-alot-of-projects)
   rm -rf contribution
   ;;
 
+imports-guava)
```
Comment (romani) [inline .ci/wercker.sh]: name should tell what it do.
name should be "no-error-imports-guava"

File: .ci/wercker.sh [hunk 2]
```diff
@@ -276,6 +276,32 @@ no-exception-alot-of-projects)
...
+  CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
+                     --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
+  echo CS_version: ${CS_POM_VERSION}
+  checkout_from https://github.com/checkstyle/contribution.git
+  cd .ci-temp/contribution/checkstyle-tester
+  groovy ./launch.groovy --checkstyleVersion ${CS_POM_VERSION} \
+    --listOfProjects checks-import-order/projects-to-test-imports-guava.properties \
+    --config checks-import-order/checks-imports-error-guava.xml
+  cd ../../
```
Comment (romani) [inline .ci/wercker.sh]: we do not need just execution .... it is not no-exception validation.
We need to make sure that there are no errors in report.
please grep `.ci-temp/contribution/checkstyle-tester/reports/guava/site/index.html` to make sure report is empty

File: wercker.yml [hunk 3]
```diff
@@ -222,6 +222,26 @@ build:
           echo "build is skipped ..."
         fi
 
+  - script:
+      name: Imports - guava ....
```
Comment (romani) [inline wercker.yml]: name should be "NoErrorTest - guava imports"

Comment (pbludov) [reply wercker.yml]: done

File: .ci/wercker.sh (hunk 2 above)
Comment (pbludov) [reply .ci/wercker.sh]: done

File: .ci/wercker.sh (hunk 1 above)
Comment (pbludov) [reply .ci/wercker.sh]: done

File: .ci/wercker.sh [hunk 4]
```diff
@@ -276,6 +276,50 @@ no-exception-alot-of-projects)
...
+  CONFIG=checks-import-order/checks-imports-error-guava.xml
+  REPORT=reports/guava/site/index.html
+  CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
+                     --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
+  echo CS_version: ${CS_POM_VERSION}
+  checkout_from https://github.com/checkstyle/contribution.git
+  cd .ci-temp/contribution/checkstyle-tester
+  groovy ./launch.groovy --listOfProjects $PROJECTS --config $CONFIG \
+      --checkstyleVersion ${CS_POM_VERSION}
+  SUMMARY=(`sed -n "s/^<td>\(.*\)<.td>.*/\1/p" < $REPORT`)
```
Comment (romani) [inline .ci/wercker.sh]: please do `grep -R -A 5 "&#160;Warning</td>" . | cat` , and do exit 1 if content is not empty. If will be good to print content before exit.

File: .ci/wercker.sh [hunk 5]
```diff
@@ -276,6 +276,50 @@ no-exception-alot-of-projects)
...
+  cd ../../
+  rm -rf contribution
+  if ((${SUMMARY[3]})) ; then
+    echo "There are ${SUMMARY[3]} errors."
+    exit 1
+  else
+    echo "Inpection did not found any problems"
```
Comment (romani) [inline .ci/wercker.sh]: any warnings

**Comment by romani** [comment]
@rnveach , @daniilyar 
jenkins is hanging:
```
Running in Durability level: MAX_SURVIVABILITY
[Pipeline] node
Still waiting to schedule task
‘rnveach-slave-lubuntu-1core-4gb-ram’ is offline
```

if node is offline, why we jenkins select it ?

File: .ci/wercker.sh (hunk 4 above)
Comment (pbludov) [reply .ci/wercker.sh]: done

File: .ci/wercker.sh (hunk 5 above)
Comment (pbludov) [reply .ci/wercker.sh]: done

File: .ci/wercker.sh [hunk 6]
```diff
@@ -276,6 +276,50 @@ no-exception-alot-of-projects)
   rm -rf contribution
   ;;
 
+no-error-imports-guava)
```
[the remark this answers is not in the document]
Comment (pbludov) [reply .ci/wercker.sh]: done

**Comment by daniilyar** [comment]
@romani, we don't have the second slave yet. I allowed builds to run on both slave and master for now to unblock builds, so it already works. @rnveach , ould you please check what happened to the slave VM?

**Comment by rnveach** [comment]
@daniilyar I posted in another thread the other day what happened. https://github.com/checkstyle/checkstyle/pull/6015#issuecomment-404148233
It has been up since then and I checked the SSH port is working. Other then that you would have to tell us what is wrong as I don't know what you installed or setup on the VM.

File: .ci/wercker.sh [hunk 7]
```diff
@@ -276,6 +276,54 @@ no-exception-alot-of-projects)
...
+  echo CS_version: ${CS_POM_VERSION}
+  checkout_from https://github.com/checkstyle/contribution.git
+  cd .ci-temp/contribution/checkstyle-tester
+  groovy ./launch.groovy --listOfProjects $PROJECTS --config $CONFIG \
+      --checkstyleVersion ${CS_POM_VERSION}
+  RESULT=`grep -A 5 "&#160;Warning</td>" $REPORT | cat`
+  cd ../../
+  rm -rf contribution
+  if [ -z "$RESULT" ]; then
+    echo "Inpection did not found any warnings"
```
Comment (rnveach) [inline .ci/wercker.sh]: `found` => `find`

File: .ci/wercker.sh [hunk 8]
```diff
@@ -276,6 +276,54 @@ no-exception-alot-of-projects)
...
```
Comment (rnveach) [inline .ci/wercker.sh]: same.

File: .ci/wercker.sh (hunk 7 above)
Comment (pbludov) [reply .ci/wercker.sh]: done

File: .ci/wercker.sh (hunk 8 above)
Comment (pbludov) [reply .ci/wercker.sh]: done

**Pull request merged**